hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/snapshot/filter/TestReclaimableFilter.java

```diff
@@ -0,0 +1,288 @@
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.ozone.om.snapshot.filter;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.eq;
+import static org.mockito.Mockito.when;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import org.apache.hadoop.hdds.utils.TransactionInfo;
+import org.apache.hadoop.hdds.utils.db.Table;
+import org.apache.hadoop.ozone.om.KeyManager;
+import org.apache.hadoop.ozone.om.OmSnapshot;
+import org.apache.hadoop.ozone.om.OmSnapshotManager;
+import org.apache.hadoop.ozone.om.OzoneManager;
+import org.apache.hadoop.ozone.om.SnapshotChainManager;
+import org.apache.hadoop.ozone.om.helpers.BucketLayout;
+import org.apache.hadoop.ozone.om.helpers.SnapshotInfo;
+import org.apache.hadoop.ozone.om.lock.IOzoneManagerLock;
+import org.apache.hadoop.ozone.om.snapshot.ReferenceCounted;
+import org.apache.hadoop.ozone.om.snapshot.SnapshotUtils;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.rocksdb.RocksDBException;
+
+/**
+ * Test class for ReclaimableFilter testing general initializing of snapshot chain.
+ */
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class TestReclaimableFilter extends TestAbstractReclaimableFilter {
+
+  @Override
+  protected ReclaimableFilter initializeFilter(OzoneManager om, OmSnapshotManager snapshotManager,
+                                               SnapshotChainManager chainManager,
+                                               SnapshotInfo currentSnapshotInfo, KeyManager km,
+                                               IOzoneManagerLock lock, int numberOfPreviousSnapshotsFromChain) {
+    return new ReclaimableFilter<Boolean>(om, snapshotManager, chainManager, currentSnapshotInfo,
+        km, lock, numberOfPreviousSnapshotsFromChain) {
+      @Override
+      protected String getVolumeName(Table.KeyValue<String, Boolean> keyValue) throws IOException {
+        return keyValue.getKey().split("/")[0];
+      }
+
+      @Override
+      protected String getBucketName(Table.KeyValue<String, Boolean> keyValue) throws IOException {
+        return keyValue.getKey().split("/")[1];
+      }
+
+      @Override
+      protected Boolean isReclaimable(Table.KeyValue<String, Boolean> keyValue) throws IOException {
+        return keyValue == null || keyValue.getValue();
+      }
+    };
+  }
+
+  List<Arguments> testReclaimableFilterArguments() {
```

Comment:
We should add a doc to specify what these parameters are.

Reply:
done